### DISCUSSION TITLE
back-to-basics-pr-validations

### DISCUSSION THREAD

**Discussion by giscus** [WorkSafeBC-Common-Engineering/Common.Engineering.Discussion#24] [Technical Blog]
# back-to-basics-pr-validations

https://wsbctechnicalblog.github.io/back-to-basics-pr-validations.html

**Comment by almguru** [comment]
I completely disagree with the recommendation of turning off CD part for pull request validation. You do need to validate PR code is deployable before the merge. So, you deploy to temporary environment, run acceptance/functional tests after deployment and now you have confidence that the same procedure will work after merge to `main` branch.  And yes - that is the same CI/CD pipeline you run for every branch and pull request.

**Reply by wpschaub** [reply]
Vladimir, no one is saying that everyone should disable the CD part for PR validation. Microsoft has a useful article on how to use classic pipelines to deploy as part of PR validation flow - Deploy pull request Artifacts with classic release pipelines (https://learn.microsoft.com/en-us/azure/devops/pipelines/release/deploy-pull-request-builds?view=azure-devops) – we have a flag that allows us to enable CD in our pipeline blueprints. In our environment, though, we are not ready for CD in our PR and we specifically need a validation build, which is why we have the conditional branch logic. What I am highlighting is that the same pipeline that does the PR validation is also the one that will be triggered by the CI after merging.

**Reply by almguru** [reply]
Thanks for clarification, Willy. I read it initially as recommendation and not as possibility. :relaxed:
